components/hab/src/cli.rs

```diff
@@ -860,9 +860,8 @@ pub fn sub_sup_run() -> App<'static, 'static> {
         // is displayed confusingly as `hab-sup`
         // see: https://github.com/kbknapp/clap-rs/blob/2724ec5399c500b12a1a24d356f4090f4816f5e2/src/app/mod.rs#L373-L394
         (usage: "hab sup run [FLAGS] [OPTIONS] [--] [PKG_IDENT_OR_ARTIFACT]")
-        (@arg LISTEN_GOSSIP: --("listen-gossip") +takes_value {valid_socket_addr}
-            "The listen address for the Gossip System Gateway. If not specified, the value will \
-            be taken from the HAB_LISTEN_GOSSIP environment variable if defined. [default: 0.0.0.0:9638]")
+              (@arg LISTEN_GOSSIP: --("listen-gossip") env(GOSSIP_LISTEN_ADDRESS_ENVVAR) default_value(&GOSSIP_DEFAULT_ADDR) {valid_socket_addr}
```

Comment:
Does this no longer need `+takes_value` if there's a `default_value`?

Also, does the generated help text reflect the fact that this can be set by an environment variable?

Reply:
Yessir `default_value` implies `takes_value(true)` https://docs.rs/clap/2.32.0/clap/struct.Arg.html#method.default_value

and yessir, clap handles docs for both the env and default value :)
```
...
--listen-gossip <LISTEN_GOSSIP>        The listen address for the Gossip System Gateway. [env:
                                       HAB_LISTEN_GOSSIP=]  [default: 0.0.0.0:9638]
...
```